### PR TITLE
Feature/#167 implement logging system

### LINKED ISSUE #167: SP: 3 Implement Logging System

### PR DESCRIPTION
**Pull request by Misha86**
Added settings for  logging system

### CODE REVIEW

File: beauty/api/models.py [hunk 1]
```diff
@@ -37,6 +43,10 @@ def create_user(self, email: str, first_name: str, password=None,
 
         user.set_password(password)
         user.save(using=self._db)
+
+        logger.info(f"User {user['first_name']} with"
```
Comment (therobotisnotatoy) [inline beauty/api/models.py]: For ease of processing logs, a user ID will be more useful to us than a name

File: beauty/api/serializers/serializers_customuser.py [hunk 2]
```diff
@@ -60,12 +66,20 @@ def validate(self, data: dict) -> dict:
         confirm_password = data.get('confirm_password')
         if password and confirm_password:
             if password != confirm_password:
+                logger.error(f"Password: Password confirmation does not match")
```
Comment (therobotisnotatoy) [inline beauty/api/serializers/serializers_customuser.py]: It seems that you can use another level of log here - Warning

File: beauty/api/serializers/serializers_customuser.py [hunk 3]
```diff
@@ -60,12 +66,20 @@ def validate(self, data: dict) -> dict:
...
+
                 raise serializers.ValidationError(
                     {"password": "Password confirmation does not match."}
                 )
         elif any([password, confirm_password]):
+
+            logger.error("Password: One of the password fields is empty")
```
Comment (therobotisnotatoy) [inline beauty/api/serializers/serializers_customuser.py]: ... same as previous

File: beauty/api/views.py [hunk 4]
```diff
@@ -89,4 +100,10 @@ def get_object(self):
                                 Q(specialist=self.kwargs['user']),
                                 id=self.kwargs['id'])
         self.check_object_permissions(self.request, obj)
-        return obj
\ No newline at end of file
+
+        user = obj.customer if self.kwargs['user'] ==\
+                               obj.customer.id else obj.specialist
+
+        logger.info(f"{obj} was got for user {user} (id={user.id})")
```
Comment (therobotisnotatoy) [inline beauty/api/views.py]: ... was received for the user ...

File: beauty/api/views.py [hunk 5]
```diff
@@ -89,4 +100,10 @@ def get_object(self):
                                 Q(specialist=self.kwargs['user']),
                                 id=self.kwargs['id'])
         self.check_object_permissions(self.request, obj)
-        return obj
\ No newline at end of file
+
+        user = obj.customer if self.kwargs['user'] ==\
```
Comment (therobotisnotatoy) [inline beauty/api/views.py]: It is possible to change to the standard variant of "if/else" so as not to break the line

File: beauty/beauty/settings.py [hunk 6]
```diff
@@ -206,3 +204,59 @@
...
+    'version': 1,
+    'disable_existing_loggers': False,
+    'formatters': {
+        'verbose': {
+            'format': '[{levelname}: {asctime}] {module} | '
+                      '{funcName}:{lineno} - {message}',
+            'style': '{',
+        },
+        'simple': {
+            'format': '{levelname} {message}',
```
Comment (therobotisnotatoy) [inline beauty/beauty/settings.py]: It is preferable to add a "timestamp" for all types of logging

File: beauty/beauty/settings.py [hunk 7]
```diff
@@ -78,18 +78,17 @@
     },
```
Comment (therobotisnotatoy) [inline beauty/beauty/settings.py:78]: Notes to the whole team: Different types of quotes are randomly used throughout the file
@Misha86 @pazuzu-ua @shevchenkoira @morento101 @YaroslavBorysko @shv833 @Serhii-Voloshyn @OleksiiDatsiuk

File: beauty/api/serializers/serializers_customuser.py (hunk 2 above)
Comment (Misha86) [reply beauty/api/serializers/serializers_customuser.py]: [logging.error()](https://docs.python.org/3/library/logging.html#logging.error), [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) or [logging.critical()](https://docs.python.org/3/library/logging.html#logging.critical) as appropriate for the specific error and application domain

[logging.warning()](https://docs.python.org/3/library/logging.html#logging.warning) if there is nothing the client application can do about the situation, but the event should still be noted

File: beauty/api/serializers/serializers_customuser.py (hunk 3 above)
Comment (Misha86) [reply beauty/api/serializers/serializers_customuser.py]: [logging.error()](https://docs.python.org/3/library/logging.html#logging.error), [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) or [logging.critical()](https://docs.python.org/3/library/logging.html#logging.critical) as appropriate for the specific error and application domain

[logging.warning()](https://docs.python.org/3/library/logging.html#logging.warning) if there is nothing the client application can do about the situation, but the event should still be noted

File: beauty/api/serializers/serializers_customuser.py (hunk 2 above)
Comment (Deer-WarLord) [reply beauty/api/serializers/serializers_customuser.py]: the if statement shows us that we expect such behavior, so
debug or info should be used (depends on the noise in long term usage)

File: beauty/api/serializers/serializers_customuser.py (hunk 3 above)
Comment (Deer-WarLord) [reply beauty/api/serializers/serializers_customuser.py]: the if statement shows us that we expect such behavior, so
debug or info should be used (depends on the noise in long term usage)

File: beauty/beauty/settings.py [hunk 8]
```diff
@@ -170,39 +168,95 @@
...
+            "level": "INFO",
+            "class": "logging.handlers.RotatingFileHandler",
+            "mode": "a",
+            "maxBytes": 15728640,  # 1024 * 1024 * 15B = 15MB
+            "backupCount": 10,
+            "filename": "logs/info.log",
+            "formatter": "verbose"
+        },
+        "mail_admins": {
+            "level": "ERROR",
```
Comment (Deer-WarLord) [inline beauty/beauty/settings.py]: I'd put email handlers for critical things

File: beauty/beauty/settings.py (hunk 7 above)
Comment (Deer-WarLord) [reply beauty/beauty/settings.py:78]: ● DEBUG: You should use this level for debugging purposes in development.
● INFO: You should use this level when something interesting—but expected—happens (e.g., a user starts a new project in a project management application).
● WARNING: You should use this level when something unexpected or unusual happens. It’s not an error, but you should pay attention to it.
● ERROR: This level is for things that go wrong but are usually recoverable (e.g., internal exceptions you can handle or APIs returning error results).
● CRITICAL: You should use this level in a doomsday scenario. The application is unusable. At this level, someone should be woken up at 2 a.m.

**Comment by Deer-WarLord** [comment]
● DEBUG: You should use this level for debugging purposes in development.
● INFO: You should use this level when something interesting—but expected—happens (e.g., a user starts a new project in a project management application).
● WARNING: You should use this level when something unexpected or unusual happens. It’s not an error, but you should pay attention to it.
● ERROR: This level is for things that go wrong but are usually recoverable (e.g., internal exceptions you can handle or APIs returning error results).
● CRITICAL: You should use this level in a doomsday scenario. The application is unusable. At this level, someone should be woken up at 2 a.m.

**Pull request merged**